### PR TITLE
Adding river-ingester package

### PR DESCRIPTION
**Pull request by pbotros**
Adding river-ingester package.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by pbotros** [comment]
Hello @conda-forge/staged-recipes , this PR is ready for review; adding a new Linux/Mac package. All tests are green. Thanks!

File: recipes/river-ingester/build.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
+  -DCMAKE_INSTALL_LIBDIR=lib \
+  -DRIVER_BUILD_INGESTER=ON \
+  -DRIVER_BUILD_TESTS=OFF \
+  -DRIVER_INSTALL=ON \
+  ${CMAKE_ARGS} \
+  ../..
+cd ingester
+make
```
Comment (ocefpaf) [inline recipes/river-ingester/build.sh:24]: Is there a `make check` or `ctest` that we can run here?

Comment (pbotros) [reply recipes/river-ingester/build.sh:24]: Unfortunately the tests currently expect a particular testing host setup (i.e. Redis running) and thus are typically run manually in the source repo . A task for the future is making our tests a bit smarter to be able to run unit tests and only the integration tests requiring Redis if available.

That said, there's at least a check later in the meta.yaml that the `river-ingester` binary is built and in the appropriate place.

File: recipes/river-ingester/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - {{ pin_subpackage("river-ingester", max_pin="x.x") }}
+  # The ingester currently doesn't build on Windows, though admittedly not
+  # much effort was put in to get this working. Redis runs on Linux/Mac
+  # only anyways, so you'll need a Linux/Mac machine in the mix regardless.
+  skip: true  # [win]
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
```
Comment (ocefpaf) [inline recipes/river-ingester/meta.yaml:24]: No need for these comments.

```suggestion
```

Comment (pbotros) [reply recipes/river-ingester/meta.yaml]: Done

**Pull request merged**